pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -527,13 +539,11 @@ func getUniqueAndSharedDNSDomain(fqdnHostname, proxyDomain string) (string, stri
 	}
 
 	if len(sharedSuffixesInReverse) == 0 {
-		return fqdnHostname, ""
+		return reverseArray(partsFQDN), nil
```

Comment:
very nit: more efficient to save the result of `strings.Split(fqdnHostname, ".")` on L529 and reuse here